server/src/main/java/org/elasticsearch/index/mapper/BlockSourceReader.java

```diff
@@ -149,6 +175,41 @@ public String toString() {
         }
     }
 
+    private static class Geometries extends BlockSourceReader {
+
+        Geometries(ValueFetcher fetcher) {
+            super(fetcher);
+        }
+
+        @Override
+        protected void append(BlockLoader.Builder builder, Object v) {
+            if (v instanceof SpatialPoint point) {
+                BytesRef wkb = new BytesRef(WellKnownBinary.toWKB(new Point(point.getX(), point.getY()), ByteOrder.LITTLE_ENDIAN));
```

Comment:
I am very confused here, do we really generate points now? Do we really need this? the idea is that we only load WKB?

Reply:
I think this was defensive coding, for the variety of possible loading we will see going forward. The source value fetcher was generating WKT (but I can change that to WKB), doc-values was generating longs, and stored fields would generate points themselves (I assumed, but have not written support for that yet).

Reply:
Lets remove defensive coding, I think we should only accept WKB here.

Reply:
++ on keeping this list as small as possible. Accepting `Object` for `append` makes this whole difficult to understand, but that's how reading `_source` works at the moment and we ain't changing it here. So folks reading this have the easiest time I think we should only handle the actual things we expect to get. Is that *just* `BytesRef`, right?

Reply:
Removed support for SpatialPoint